Objects/setobject.c

```diff
@@ -631,34 +638,40 @@ set_merge(PySetObject *so, PyObject *otherset)
     return 0;
 }
 
+/*[clinic input]
+set.pop
+
+Remove and return an arbitrary set element.
+
+Raises KeyError if the set is empty.
+[clinic start generated code]*/
+
 static PyObject *
-set_pop(PySetObject *so, PyObject *Py_UNUSED(ignored))
+set_pop_impl(PySetObject *self)
+/*[clinic end generated code: output=bf7ef01d2a364703 input=4a02ec3ab60c88fb]*/
 {
     /* Make sure the search finger is in bounds */
-    setentry *entry = so->table + (so->finger & so->mask);
-    setentry *limit = so->table + so->mask;
+    setentry *entry = self->table + (self->finger & self->mask);
+    setentry *limit = self->table + self->mask;
```

Comment:
It would be nice if we could keep the diff down and not have to change all the self parameter names.

Take a look at the approach done in the recent Argument Clinic adaption of `collections.deque` and see if you can apply it here also :)

Reply:
Yes, definitely.  Arg clinic conversion should only affect the calling code and not the stable implementation code.

Reply:
I updated the PR based on the example from `collections.deque`, let me know if it's ok like this :)

Reply:
> I updated the PR based on the example from `collections.deque`, let me know if it's ok like this :)

Perfect; thanks!